programmers/6일차/모의고사.js

```diff
@@ -0,0 +1,34 @@
+function solution(answers) {
+    const patternNo1 = [1,2,3,4,5];
+    const patternNo2 = [2,1,2,3,2,4,2,5];
+    const patternNo3 = [3,3,1,1,2,2,4,4,5,5];
+    let maxNum = 0
+    let result = [];
+    let countNo1 = 0;
+    let countNo2 = 0;
+    let countNo3 = 0;
+
+    for (let index = 0; index < answers.length; index++) {
+        if (patternNo1[index % 5]  === answers[index]) countNo1++;
+        if (patternNo2[index % 8]  === answers[index]) countNo2++;
+        if (patternNo3[index % 10]  === answers[index]) countNo3++;
+    }
+    maxNum = Math.max(countNo1,countNo2,countNo3)    
+    if (maxNum === countNo1) result.push(1);
+    if (maxNum === countNo2) result.push(2)
+    if (maxNum === countNo3) result.push(3)
+    
+    return result;
+}  
+console.log(solution([1,3,2,4,2]));
+
```

Comment:
```suggestion
function solution(answers) {
    const numberOne = [1,2,3,4,5];
    const numberTwo = [2,1,2,3,2,4,2,5];
    const numberThree = [3,3,1,1,2,2,4,4,5,5];
    const result = [0,0,0];
    const answer = [];
    
    let count = 0;
    let max = 0;
    
    for (let i = 0; i < answers.length; i++) {
        if (numberOne[i % 5] === answers[i]) result[0] += 1;
        if (numberTwo[i % 8] === answers[i]) result[1] += 1;
        if (numberThree[i % 10] === answers[i]) result[2] += 1;
    }
    
    max = Math.max(...result);
    
    result.forEach((el,idx) => {
        if (el === max) {
            answer.push(idx + 1);
    }})
    
    return answer;
    
}  
console.log(solution([1,3,2,4,2]));

```
잘 푸셨어요 같은 방식인데  이렇게 하면 조금 더 간결하게 풀 수 있을 것 같네요